spec.md

```diff
@@ -2709,6 +2713,67 @@ need only respect the 2-state behavior of a circuit.  This is a limitation on
 the scope of what behavior is observable (i.e., a relaxation of the
 ["as-if"](https://en.wikipedia.org/wiki/As-if_rule) rule).
 
+## Indeterminate Values
+
+An indeterminate value represents a value which is unknown or unspecified.  
+Indeterminate values are generally implementation defined, with constraints 
+specified below.  An indeterminate value may be assumed to be any specific 
+value, at an implementation's discretion, if, in doing so, all observable 
+behavior is as if the indeterminate value always took the specific value.
+
+This allows transformations such as the following, where when `a` has an 
+indeterminate value, the implementation chooses to consistently give it a value 
+of 4.  There is no visibility of `a` when it has an indeterminate value which 
+does not see it as 4.
+``` firrtl
+module IValue :
+  output o : UInt<8>
+  input c : UInt<1>
+
+  wire a : UInt<8>
+  a is invalid
+  when c :
+    a <= UInt<3>("h4")
+  o <= a
+```
+is transformed to:
+``` firrtl
+module IValue :
+  output o : UInt<8>
+  input c : UInt<1>
+
+  o <= UInt<3>("h4")
+```
+Note that it is equally correct to produce:
+``` firrtl
+module IValue :
+  output o : UInt<8>
+  input c : UInt<1>
+
+  wire a : UInt<8>
+  when c :
+    a <= UInt<3>("h4")
+   else :
+     a <= UInt<3>("h42")
+  o <= a
+```
+
+The behavior of constructs which cause indeterminate values is implementation 
+defined with the following constraints.  
+* Register initialization is done in a consistent way for all registers.  If 
+code is generated to randomly initialize some registers (or 0 fill them, etc), 
+it should be generated for all registers.
```

Comment:
Why is that a constraint? What if someone (for verification purposes) wants to randomize only some registers. Since their startup value is indeterminate, any concrete assignment should be valid.

Reply:
In this case, an implementation would provide a way to initialize some registers, but not others.  But that mechanism would be applied uniformly.